sdks/typescript/client/src/components/AppRenderer.tsx

```diff
@@ -0,0 +1,389 @@
+import { useEffect, useRef, useState } from "react";
+
+import { Client } from "@modelcontextprotocol/sdk/client/index.js";
+import {
+  type CallToolResult,
+  type LoggingMessageNotification,
+  McpError,
+  ErrorCode,
+} from "@modelcontextprotocol/sdk/types.js";
+
+import {
+  AppBridge,
+  PostMessageTransport,
+} from "@modelcontextprotocol/ext-apps/app-bridge";
+
+import {
+  getToolUiResourceUri,
+  readToolUiResourceHtml,
+  setupSandboxProxyIframe,
+} from "../utils/app-host-utils";
+import { UIActionResult } from "..";
+
+/**
+ * Props for the AppRenderer component.
+ */
+export interface AppRendererProps {
+  /** URL to the sandbox proxy HTML that will host the tool UI iframe */
+  sandboxProxyUrl: URL;
```

Comment:
Perhaps we should change it to a `sandbox` prop with an internal object for `url`. I imagine `sandbox` will have additional properties in the future (such as permissions). WDYT?